src/de/mod.rs

```diff
@@ -1018,6 +1018,39 @@ mod tests {
         assert_eq!(serde_json::from_str::<Nothing>(r#"null"#).unwrap(), Nothing);
     }
 
+    #[test]
+    fn struct_with_flatten() {
+        #[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+        struct Pagination {
+            limit: u64,
+            offset: u64,
+            total: u64,
+        }
+
+        #[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
```

Comment:
This is a great test case. But it would be good to have one or two for the specific version we would use in CosmWasm.

```rust
struct ExecMsg1 {
  something: String,
}

struct ExecMsg2 {
  other: u64,
}

/// one example to test
enum ExecMsg {
  #[serde(flatten)]
  One(ExecMsg1),
  #[serde(flatten)]
  Two(ExecMsg2),
}

/// another example to test
enum ExecMsg {
  #[serde(flatten)]
  One(ExecMsg1),
  // important to leave top level along with a flattened variant
  SetAge { age: u32 },
  // also interesting to see if we can use this format with them (unflattened) 
  Custom(CustomData),
}
```

You can make it a bit nicer than that, but combining a cw20 ExecMsg and my contracts custom one is a clear case.

Reply:
Yeah, right. I can easily make the flatten test for enums instead of structs. This is the part we need for CosmWasm.

Reply:
Turns out flatten is not for enums (see [feature request](https://github.com/serde-rs/serde/issues/1402)).

So this PR would close the ticket and if we hit new issues, we can create more refined ones.

Reply:
Great link.
Yes, we need untagged for that, which is another issue https://github.com/CosmWasm/serde-json-wasm/issues/43

This PR closes the linked issue well